modules/tinymce/src/core/test/ts/browser/keyboard/ArrowKeysInlineBoundariesTest.ts

```diff
@@ -336,4 +336,43 @@ describe('browser.tinymce.core.keyboard.ArrowKeysInlineBoundariesTest', () => {
       TinyAssertions.assertCursor(editor, [ 0, 0 ], 0);
     });
   });
+
+  context('Arrow keys when selection is not collapsed', () => {
+    it('TINY-8601: should move caret before selected inline boundary node when arrow left', () => {
+      const editor = hook.editor();
+      editor.setContent('<p>test <span class="mce-annotation">span</span> selection</p>');
+      TinySelections.setSelection(editor, [ 0 ], 1, [ 0 ], 2, true);
+      TinyContentActions.keystroke(editor, Keys.left());
+      TinyAssertions.assertCursor(editor, [ 0, 0 ], 5);
+      assertCaretAtZwsp(editor);
+    });
+
+    it('TINY-8601: should move caret after selected inline boundary node when arrow right', () => {
+      const editor = hook.editor();
+      editor.setContent('<p>test <span class="mce-annotation">span</span> selection</p>');
+      TinySelections.setSelection(editor, [ 0 ], 1, [ 0 ], 2, true);
+      TinyContentActions.keystroke(editor, Keys.right());
+      TinyAssertions.assertCursor(editor, [ 0, 2 ], 1);
+      assertCaretAfterZwsp(editor);
+    });
+
+    it('TINY-8601: should collapse inner text selection to the start when arrow left', () => {
+      const editor = hook.editor();
+      editor.setContent('<p>test <span class="mce-annotation">span</span> selection</p>');
+      TinySelections.setSelection(editor, [ 0, 1, 0 ], 0, [ 0, 1, 0 ], 4, true);
+      TinyContentActions.keystroke(editor, Keys.left());
+      TinyAssertions.assertCursor(editor, [ 0, 1, 0 ], 1);
```

Comment:
This assertion isn't right... it should be collapsing the selection or moving it outside the span, not moving it one position to the right (especially when we press the left arrow key).

Reply:
Oh wait... is this the ZWSP text node that it's on? If so that'd make some sense, in which case can you please add a comment to clarify that.

Reply:
I've added the comment

Reply:
Thankyou!